xc/device.ml

```diff
@@ -2417,16 +2421,19 @@ module Backend = struct
             ; (global |> List.map (fun x -> ["-global"; x]) |> List.concat)
             ; (info.Dm_Common.parallel |> function None -> [ "-parallel"; "null"] | Some x -> [ "-parallel"; x])
             ; qmp
-            ; pv_device
             ; xen_platform_device
             ] in
 
         (* Sort the VIF devices by devid *)
-        let nics = List.stable_sort (fun (_,_,a) (_,_,b) -> compare a b) info.Dm_Common.nics in
-        if List.length nics > Dm_Common.max_emulated_nics then
-          debug "Limiting the number of emulated NICs to %d" Dm_Common.max_emulated_nics;
+        let nics = List.stable_sort
+          (fun (_,_,a) (_,_,b) -> compare a b) info.Dm_Common.nics in
+        let nic_count = List.length nics in
+        let nic_max   = Dm_Common.max_emulated_nics in
+        if nic_count > nic_max then
+          debug "Limiting the number of emulated NICs to %d" nic_max;
         (* Take the first 'max_emulated_nics' elements from the list. *)
-        let nics = Xapi_stdext_std.Listext.List.take Dm_Common.max_emulated_nics nics in
+        let nics = Xapi_stdext_std.Listext.List.take nic_max nics in
+        let nic_count = min nic_count nic_max in (* update count *)
```

Comment:
Perhaps you could change less code by just adding this after the list has been filtered:

    let nic_count = List.length nics in

It doesn't matter either way.

Reply:
I thought about this too and I agree it would work as well. I made most changes to get rid of the long lines and names.